torch/_dynamo/utils.py

```diff
@@ -241,7 +242,8 @@ def time_wrapper(*args, **kwargs):
                 compilation_time_metrics[key] = []
             with torch.profiler.record_function(f"{key} (dynamo_timed)"):
                 t0 = time.time()
-                r = func(*args, **kwargs)
+                with _temp_pop_torch_function_mode():
```

Comment:
It seems weird to unconditionally have this logic be aware of "whatever the top torch_function mode is"